src/odemis/acq/stitching/_weaver.py

```diff
@@ -43,66 +44,233 @@ def collageWeaver(tiles, border=None):
+    
+    def getFullImage(self):
+        """
+        return (2D DataArray): same dtype as the tiles, with shape corresponding to the bounding box. 
+        """
+             
+        # Sort the tiles by time, to avoid random order in "Z", and high-light the
+        # acquisition order.
+        tiles = sorted(self.tiles, key=lambda t: t.metadata.get(model.MD_ACQ_DATE, 0))
+       
+        # Merge the correction metadata inside each image (to keep the rest of the
+        # code simple)
+
+        tiles = [model.DataArray(t, t.metadata.copy()) for t in tiles]
+        
+        for t in tiles:
+            img.mergeMetadata(t.metadata)
+
+        # Compute the bounding box of each tile and the global bounding box
+    
+        # Get a fixed pixel size by using the first one
+        # TODO: use the mean, in case they are all slightly different due to correction?
+        pxs = tiles[0].metadata[model.MD_PIXEL_SIZE]
+        
+        tbbx_phy = []  # tuples of ltrb in physical coordinates
+        for t in tiles:
+            c = t.metadata[model.MD_POS]
+            w = t.shape[-1], t.shape[-2]
+            if not util.almost_equal(pxs[0], t.metadata[model.MD_PIXEL_SIZE][0], rtol=0.01):
+                logging.warning("Tile @ %s has a unexpected pixel size (%g vs %g)",
+                                c, t.metadata[model.MD_PIXEL_SIZE][0], pxs[0])
+            bbx = (c[0] - (w[0] * pxs[0] / 2), c[1] - (w[1] * pxs[1] / 2),
+                   c[0] + (w[0] * pxs[0] / 2), c[1] + (w[1] * pxs[1] / 2))
+            
+            tbbx_phy.append(bbx)
+    
+        gbbx_phy = (min(b[0] for b in tbbx_phy), min(b[1] for b in tbbx_phy),
+                    max(b[2] for b in tbbx_phy), max(b[3] for b in tbbx_phy))
+    
+        # Compute the bounding-boxes in pixel coordinates
+        tbbx_px = []
+        
+        glt = gbbx_phy[0], gbbx_phy[3]  # that's the origin (Y is max as Y is inverted)
+        for bp, t in zip(tbbx_phy, tiles):
+            lt = (int((bp[0] - glt[0]) / pxs[0]), int(-(bp[3] - glt[1]) / pxs[1]))
+            w = t.shape[-1], t.shape[-2]
+            bbx = (lt[0], lt[1],
+                   lt[0] + w[0], lt[1] + w[1])
+            tbbx_px.append(bbx)
+   
+        gbbx_px = (min(b[0] for b in tbbx_px), min(b[1] for b in tbbx_px),
+                   max(b[2] for b in tbbx_px), max(b[3] for b in tbbx_px))
+    
+        assert gbbx_px[0] == gbbx_px[1] == 0
+       
+        # TODO: warn if the global area is much bigger than the sum of the tile area
+    
+        # Paste each tile
+        logging.debug("Generating global image of size %dx%d px", gbbx_px[-2], gbbx_px[-1])
+        im = numpy.empty((gbbx_px[-1], gbbx_px[-2]), dtype=tiles[0].dtype)
+        # TODO: what value to use for background? Use minimum of the values in the tiles
+        im[:] = numpy.amin(tiles)
+       
+        if len(tiles) > 1:
+            # Overlap
+            diff = max(tiles[1].metadata[model.MD_POS][0]-tiles[0].metadata[model.MD_POS][0],
```

Comment:
The overlap is not the same between each tile (although, generally, they should be similar). So you cannot rely only on the distance between the first and second tile.